core/src/main/java/org/apache/carbondata/core/locks/CarbonLockFactory.java

```diff
@@ -52,23 +52,23 @@
    */
   public static ICarbonLock getCarbonLockObj(AbsoluteTableIdentifier absoluteTableIdentifier,
       String lockFile) {
-    switch (lockTypeConfigured) {
-      case CarbonCommonConstants.CARBON_LOCK_TYPE_LOCAL:
```

Comment:
Why swith case change to if else

Reply:
This change is to support multiple filesystems on a single carbonsession. So at a time if you work on S3 and local together, it should implicitly understand what lock type it should provide. 

So we are identifying lock type based on the type of file and the conditional match falls on two variables:  one is lockTypeConfigured and second, we have to identify what type of file it is?(whether s3, hdfs, local) on the basis of which we will set lockTypeConfigured.

Due to this change now we don't need to specify LOCK_TYPE in carbonProperties explicitly for s3, hdfs and local.